spackbot/pr_add_reviewers.py

```diff
@@ -0,0 +1,235 @@
+            if not await found(gh.getitem(collaborators_url, {"collaborator": user})):
+                non_reviewers.append(user)
+                continue
+
+            result = await gh.getitem(
+                collaborators_url + "/permission",
+                {"collaborator": user},
+            )
+            level = result["permission"]
+            logger.debug(f"Permission level: {level}")
+            reviewers.append(user)
+
+        # If they have permission, add them
+        # https://docs.github.com/en/rest/reference/pulls#request-reviewers-for-a-pull-request
+        if reviewers:
+            logger.debug(f"Requesting review from: {reviewers}")
+
+            # There is a limit of 15 reviewers, so take the first 15
+            await gh.post(
+                pull_request["url"] + "/requested_reviewers",
+                {},
+                data={"reviewers": reviewers[:15]},
+            )
+
+        # If not, give them permission and comment
+        if non_reviewers:
+            logger.debug(f"Adding collaborators: {non_reviewers}")
+
+            # We do not want to give users write permission here, as write
+            # permission allows people to submit approving reviews for
+            # auto-merge. Instead, we'd like to give them triage, which allows
+            # them to be requested, label the PR, etc., but not determine
+            # whether it should be merged.
+            # https://docs.github.com/en/rest/reference/repos#add-a-repository-collaborator
+            for user in non_reviewers:
+                await gh.put(
+                    repository["collaborators_url"],
+                    {"collaborator": user},
+                    # TODO: Looks like the collaborators API does not yet support
+                    # 'triage' here, so we use 'read' instead.
+                    data={"permission": "read"},
+                )
+
+            # https://docs.github.com/en/rest/reference/issues#create-an-issue-comment
+            comment_body = non_reviewers_comment.format(
+                packages_with_maintainers="\n* ".join(sorted(maintained_pkgs)),
+                non_reviewers=" @".join(sorted(non_reviewers)),
+            )
+            await gh.post(pull_request["comments_url"], {}, data={"body": comment_body})
+
+        # Ask people to maintain packages that don't have maintainers.
+        if unmaintained_pkgs:
+            # Ask for maintainers
+            # https://docs.github.com/en/rest/reference/issues#create-an-issue-comment
+            comment_body = no_maintainers_comment.format(
+                author=pull_request["user"]["login"],
+                packages_without_maintainers="\n* ".join(sorted(unmaintained_pkgs)),
+            )
+            await gh.post(pull_request["comments_url"], {}, data={"body": comment_body})
+
+
+@router.register("pull_request", action="opened")
+@router.register("pull_request", action="synchronize")
```

Comment:
Should this be dropped before merging? We want to add labels on synch but not re-ping the same maintainers.

Reply:
So the thing here is that they may modify more files in the meantime.  What do you want to do about that?

Reply:
My thoughts:

1. If we can request review instead of comment on PR, then synchronize is fine
2. If we can only comment on PR, we don't want to comment after every single commit

Reply:
ok, how about we remove it for now, and add that functionality later.

Reply:
I have this feeling we'll need to iterate anyway so I just want to get this thing live.

Reply:
Completely agree